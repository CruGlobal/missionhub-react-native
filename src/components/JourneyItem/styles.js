
import { StyleSheet } from 'react-native';
import theme from '../../theme';

export default StyleSheet.create({
  row: {
    paddingVertical: 19,
    paddingHorizontal: 24,
    backgroundColor: theme.white,
    width: theme.fullWidth,
  },
  icon: {
    color: theme.secondaryColor,
    paddingRight: 15,
  },
  textWrap: {
    borderLeftColor: theme.separatorColor,
    borderLeftWidth: 1,
    paddingLeft: 20,
    paddingBottom: 5,
  },
  date: {
    color: theme.inactiveColor,
    fontSize: 10,
  },
  text: {
    fontSize: 14,
  },
});
