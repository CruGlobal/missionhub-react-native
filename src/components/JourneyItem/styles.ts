import { StyleSheet } from 'react-native';

import theme from '../../theme';

export default StyleSheet.create({
  row: {
    paddingVertical: 19,
    paddingRight: 24,
    backgroundColor: theme.white,
    width: theme.fullWidth,
  },
  icon: {
    fontSize: 40,
    textAlign: 'center',
    alignSelf: 'center',
    color: theme.primaryColor,
  },
  commentIcon: {
    fontSize: 32,
  },
  textWrap: {
    borderLeftColor: theme.separatorColor,
    borderLeftWidth: 1,
    paddingLeft: 20,
    paddingBottom: 5,
  },
  date: {
    ...theme.textRegular10,
    color: theme.lightGrey,
  },
  title: {
    ...theme.textRegular16,
    color: theme.primaryColor,
  },
  text: theme.textRegular14,
  question: {
    ...theme.textRegular14,
    marginTop: 8,
    color: theme.lightGrey,
  },
});
