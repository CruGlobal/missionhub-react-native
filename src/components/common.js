// Common components that we can import with simple destructuring syntax
// So when one of these components are loaded in, all files will be called also
import Button from './Button';
import DateComponent from './DateComponent';
import Flex from './Flex';
import Icon from './Icon';
import Input from './Input';
import Loading from './Loading';
import Text from './Text';
import Touchable from './Touchable';
import Separator from './Separator';
import RefreshControl from './RefreshControl';

export {
  Button,
  DateComponent,
  Flex,
  Icon,
  Input,
  Loading,
  Text,
  Touchable,
  Separator,
  RefreshControl,
};
