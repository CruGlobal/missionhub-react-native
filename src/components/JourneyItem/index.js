import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { translate } from 'react-i18next';

import { Flex, Text, Icon, DateComponent } from '../common';
import styles from './styles';
import { INTERACTION_TYPES } from '../../constants';

const interactionsArr = Object.keys(INTERACTION_TYPES).map((key) => INTERACTION_TYPES[key]);

@translate('journeyItem')
export default class JourneyItem extends Component {
  setNativeProps(nProps) { this._view.setNativeProps(nProps); }
  renderDate() {
    const { item } = this.props;

    return <DateComponent date={item.date} style={styles.date} format="LL" />;
  }

  hasOldStage(item) {
    return item.old_pathway_stage.name;
  }

  translatableStage(item) {
    return { personName: item.personName, oldStage: item.old_pathway_stage.name, newStage: item.new_pathway_stage.name };
  }

  renderTitle() {
    const { t, item, type } = this.props;
    let title;
    if (type === 'step') {
      title = t('stepTitle');
    } else if (type === 'stage') {
      if (item.old_pathway_stage.name) {
        title = t('stageTitle', this.translatableStage(item));
      } else {
        title = item.new_pathway_stage.name;
      }
    } else if (type === 'survey' && item.survey) {
      title = item.survey.title;
    } else if (type === 'interaction') {
      const interaction = interactionsArr.find((i) => i.id === item.interaction_type_id);
      if (interaction) {
        title = t(interaction.translationKey);
      }
    }

    if (!title) return null;

    return (
      <Text style={styles.title}>
        {title}
      </Text>
    );
  }
  renderText() {
    const { t, item, type } = this.props;
    let text;
    if (type === 'step') {
      text = item.title;
      if (item.note) {
        text = `${text}\n\n${item.note}`;
      }
    } else if (type === 'stage') {
      if (this.hasOldStage(item)) {
        text = t('stageText', this.translatableStage(item));
      } else {
        text = t('stageStart', this.translatableStage(item));
      }
    } else {
      text = item.text;
    }

    return (
      <Text style={styles.text}>
        {text}
      </Text>
    );
  }

  renderIcon() {
    const { item, type } = this.props;
    let iconType;
    if (type === 'step') {
      iconType = 'stepsIcon';
    } else if (type === 'stage') {
      iconType = 'journeyIcon';
    } else if (type === 'survey') {
      iconType = 'surveyIcon';
    } else if (type === 'interaction') {
      const interaction = interactionsArr.find((i) => i.id === item.interaction_type_id);
      if (interaction) {
        iconType = interaction.iconName;
      }
    }

    if (!iconType) return null;

    return (
      <Icon
        name={iconType}
        type="MissionHub"
        size={32}
        style={styles.icon}
      />
    );
  }

  renderContent() {
    if (this.props.type === 'survey') return this.renderSurvey();
    return (
      <Flex value={1} direction="column" style={styles.textWrap}>
        {this.renderDate()}
        {this.renderTitle()}
        {this.renderText()}
      </Flex>
    );
  }

  renderSurvey() {
    const { answers } = this.props.item;
    return (
      <Flex value={1} direction="column" style={styles.textWrap}>
        {this.renderDate()}
        {this.renderTitle()}
        {
          answers.map((a) => (
            <Flex direction="column" key={a.id}>
              <Text style={styles.question}>{a.question.label}</Text>
              <Text style={styles.text}>{a.value}</Text>
            </Flex>
          ))
        }
      </Flex>
    );
  }

  render() {
    const { type } = this.props;
    return (
      <Flex
        ref={(c) => this._view = c}
        direction="row"
        align="center"
        style={[
          styles.row,
          type && styles[type] ? styles[type] : null,
        ]}
      >
        {this.renderIcon()}
        {this.renderContent()}
      </Flex>
    );
  }

}

JourneyItem.propTypes = {
  item: PropTypes.shape({
    id: PropTypes.string.isRequired,
    text: PropTypes.string,
    title: PropTypes.string,
    completed_at: PropTypes.date,
  }).isRequired,
  type: PropTypes.oneOf([ 'step', 'stage', 'survey', 'interaction' ]),
};
