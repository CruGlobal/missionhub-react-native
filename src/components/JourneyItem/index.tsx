import React, { Component } from 'react';
import PropTypes from 'prop-types';
import i18next from 'i18next';
import { withTranslation } from 'react-i18next';
import { Text } from 'react-native';

import { Flex, Icon, DateComponent } from '../common';
import { INTERACTION_TYPES, ACCEPTED_STEP } from '../../constants';
import { getAssignedByName, getAssignedToName } from '../../utils/common';
import { localizedStageSelector } from '../../selectors/stages';

import styles from './styles';

const interactionsArr = Object.keys(INTERACTION_TYPES).map(
  // @ts-ignore
  key => INTERACTION_TYPES[key],
);

// @ts-ignore
@withTranslation('journeyItem')
export default class JourneyItem extends Component {
  // @ts-ignore
  setNativeProps(nProps) {
    // @ts-ignore
    this._view.setNativeProps(nProps);
  }

  renderDate() {
    // @ts-ignore
    const { item } = this.props;
    return (
      <DateComponent
        date={
          item._type === ACCEPTED_STEP ? item.completed_at : item.created_at
        }
        style={styles.date}
        format="LL"
      />
    );
  }

  // @ts-ignore
  oldStage(item) {
    return localizedStageSelector(item.old_pathway_stage, i18next.language)
      .name;
  }

  // @ts-ignore
  translatableStage(item) {
    return {
      personName: item.person.first_name,
      oldStage: this.oldStage(item),
      newStage: localizedStageSelector(item.new_pathway_stage, i18next.language)
        .name,
    };
  }

  // @ts-ignore
  isSelfPathwayProgressionAudit(item) {
    // @ts-ignore
    return this.props.myId === item.person.id;
  }

  renderTitle() {
    const {
      // @ts-ignore
      t,
      // @ts-ignore
      item,
      // @ts-ignore
      myId,
      // @ts-ignore
      personFirstName,
      // @ts-ignore
      item: { _type },
    } = this.props;
    let title;
    if (_type === ACCEPTED_STEP) {
      const pathwayStage =
        (item.challenge_suggestion &&
          item.challenge_suggestion.pathway_stage &&
          ` ${
            localizedStageSelector(
              item.challenge_suggestion.pathway_stage,
              i18next.language,
            ).name
          } `) ||
        ' ';
      title = t('stepTitle', { stageName: pathwayStage });
    } else if (_type === 'pathway_progression_audit') {
      if (this.oldStage(item)) {
        title = t('stageTitle', this.translatableStage(item));
      } else {
        title = localizedStageSelector(item.new_pathway_stage, i18next.language)
          .name;
      }
    } else if (_type === 'answer_sheet' && item.survey) {
      title = item.survey.title;
    } else if (_type === 'interaction') {
      const interaction = interactionsArr.find(
        i => i.id === `${item.interaction_type_id}`,
      );
      if (interaction) {
        title = t(interaction.translationKey);
      }
    } else if (_type === 'contact_assignment') {
      const assignedToName = getAssignedToName(myId, item);
      const assignedByName = getAssignedByName(myId, item);

      title = t('contactAssignment', {
        assignedByName,
        assignedContactName: personFirstName,
        assignedToName,
      });
    } else if (_type === 'contact_unassignment') {
      const assignedToName = getAssignedToName(myId, item);

      title = t('contactUnassignment', {
        assignedContactName: personFirstName,
        assignedToName,
      });
    }

    if (!title) {
      return null;
    }

    return <Text style={styles.title}>{title}</Text>;
  }

  renderText() {
    const {
      // @ts-ignore
      t,
      // @ts-ignore
      item,
      // @ts-ignore
      item: { _type },
    } = this.props;
    let text;
    if (_type === ACCEPTED_STEP) {
      text = item.title;
      if (item.note) {
        text = `${text}\n\n${item.note}`;
      }
    } else if (_type === 'pathway_progression_audit') {
      if (this.oldStage(item)) {
        if (this.isSelfPathwayProgressionAudit(item)) {
          text = t('stageTextSelf', this.translatableStage(item));
        } else {
          text = t('stageText', this.translatableStage(item));
        }
      } else {
        if (this.isSelfPathwayProgressionAudit(item)) {
          text = t('stageStartSelf', this.translatableStage(item));
        } else {
          text = t('stageStart', this.translatableStage(item));
        }
      }
    } else if (_type === 'interaction') {
      text = item.comment;
    }

    if (!text) {
      return null;
    }

    return <Text style={styles.text}>{text}</Text>;
  }

  renderIcon() {
    const {
      // @ts-ignore
      item,
      // @ts-ignore
      item: { _type },
    } = this.props;
    let iconType;
    if (_type === ACCEPTED_STEP) {
      iconType = 'stepsIcon';
    } else if (_type === 'pathway_progression_audit') {
      iconType = 'journeyIcon';
    } else if (_type === 'answer_sheet') {
      iconType = 'surveyIcon';
    } else if (_type === 'interaction') {
      const interaction = interactionsArr.find(
        i => i.id === `${item.interaction_type_id}`,
      );
      if (interaction) {
        iconType = interaction.iconName;
      }
    } else if (
      _type === 'contact_assignment' ||
      _type === 'contact_unassignment'
    ) {
      iconType = 'statusIcon';
    }

    if (!iconType) {
      return null;
    }

    return (
      <Flex value={1}>
        <Icon
          name={iconType}
          type="MissionHub"
          size={32}
          style={[
            styles.icon,
            iconType === 'commentIcon' ? styles.commentIcon : {},
          ]}
        />
      </Flex>
    );
  }

  renderContent() {
    // @ts-ignore
    if (this.props.item._type === 'answer_sheet') {
      return this.renderSurvey();
    }
    return (
      <Flex value={3.5} direction="column" style={styles.textWrap}>
        {this.renderDate()}
        {this.renderTitle()}
        {this.renderText()}
      </Flex>
    );
  }

  renderSurvey() {
    // @ts-ignore
    const { answers } = this.props.item;
    return (
      <Flex value={3.5} direction="column" style={styles.textWrap}>
        {this.renderDate()}
        {this.renderTitle()}
        {/* 
        // @ts-ignore */}
        {answers.map(a => (
          <Flex direction="column" key={a.id}>
            <Text style={styles.question}>{a.question.label}</Text>
            <Text style={styles.text}>{a.value}</Text>
          </Flex>
        ))}
      </Flex>
    );
  }

  // @ts-ignore
  ref = c => (this._view = c);

  render() {
    return (
      <Flex ref={this.ref} direction="row" align="center" style={styles.row}>
        {this.renderIcon()}
        {this.renderContent()}
      </Flex>
    );
  }
}

// @ts-ignore
JourneyItem.propTypes = {
  item: PropTypes.shape({
    id: PropTypes.string.isRequired,
    _type: PropTypes.oneOf([
      ACCEPTED_STEP,
      'pathway_progression_audit',
      'answer_sheet',
      'interaction',
      'contact_assignment',
      'contact_unassignment',
    ]),
    text: PropTypes.string,
    title: PropTypes.string,
    // @ts-ignore
    completed_at: PropTypes.date,
  }).isRequired,
  myId: PropTypes.string.isRequired,
  personFirstName: PropTypes.string.isRequired,
};
