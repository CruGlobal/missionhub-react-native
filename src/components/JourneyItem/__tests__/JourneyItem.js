import 'react-native';
import React from 'react';

import { getAssignedByName, getAssignedToName } from '../../../utils/common';

import JourneyItem from '..';

import { testSnapshotShallow, renderShallow } from '../../../../testUtils';

jest.mock('../../../utils/common');

const myId = '484893';
const person = {
  id: '889433',
  _type: 'person',
  first_name: 'Test Person',
};
const date = '2017-12-06T14:24:52Z';

beforeEach(() => {
  getAssignedToName.mockReturnValue('Roger');
  getAssignedByName.mockReturnValue('Billy');
});

describe('step', () => {
  const mockStep = {
    id: '1',
    _type: 'accepted_challenge',
    title: 'Test Step',
    completed_at: date,
  };
  const challenge_suggestion = {
    pathway_stage: {
      name: 'Guiding',
    },
  };

  it('is rendered correctly without comment', () => {
    testSnapshotShallow(
      <JourneyItem
        item={mockStep}
        myId={myId}
        personFirstName={person.first_name}
      />,
    );
  });

  it('is rendered correctly with comment', () => {
    testSnapshotShallow(
      <JourneyItem
        item={{ ...mockStep, note: 'test comment on completed step' }}
        myId={myId}
        personFirstName={person.first_name}
      />,
    );
  });

  it('is rendered correctly with pathway stage', () => {
    testSnapshotShallow(
      <JourneyItem
        item={{ ...mockStep, challenge_suggestion }}
        myId={myId}
        personFirstName={person.first_name}
      />,
    );
  });
});

describe('stage', () => {
  const mockStageProgression = {
    id: '3',
    _type: 'pathway_progression_audit',
    comment: 'Test Stage Change',
    old_pathway_stage: {
      id: '1',
      _type: 'pathway_stage',
      name: 'Uninterested',
    },
    new_pathway_stage: { id: '2', _type: 'pathway_stage', name: 'Curious' },
    created_at: date,
    person,
  };

  it('is rendered correctly with old stage for a contact', () => {
    testSnapshotShallow(
      <JourneyItem
        item={mockStageProgression}
        myId={myId}
        personFirstName={person.first_name}
      />,
    );
  });

  it('is rendered correctly with old stage for self', () => {
    testSnapshotShallow(
      <JourneyItem
        item={mockStageProgression}
        myId={person.id}
        personFirstName={person.first_name}
      />,
    );
  });

  it('is rendered correctly without old stage for a contact', () => {
    testSnapshotShallow(
      <JourneyItem
        item={{
          ...mockStageProgression,
          old_pathway_stage: {},
        }}
        myId={myId}
        personFirstName={person.first_name}
      />,
    );
  });

  it('is rendered correctly without old stage for self', () => {
    testSnapshotShallow(
      <JourneyItem
        item={{
          ...mockStageProgression,
          old_pathway_stage: {},
        }}
        myId={person.id}
        personFirstName={person.first_name}
      />,
    );
  });
});

it('renders survey correctly', () => {
  testSnapshotShallow(
    <JourneyItem
      item={{
        id: '5',
        _type: 'answer_sheet',
        survey: { title: 'Test Survey' },
        answers: [
          { id: '1', question: { label: 'Question 1' }, value: 'Answer 1' },
          { id: '2', question: { label: 'Question 2' }, value: 'Answer 2' },
          { id: '3', question: { label: 'Question 3' }, value: 'Answer 3' },
        ],
        created_at: date,
      }}
      myId={myId}
      personFirstName={person.first_name}
    />,
  );
});

it('renders interaction correctly', () => {
  testSnapshotShallow(
    <JourneyItem
      item={{
        id: '4',
        _type: 'interaction',
        interaction_type_id: 1,
        comment: 'Test Interaction',
        initiators: [{ id: myId }],
        organization: null,
        created_at: date,
      }}
      myId={myId}
      personFirstName={person.first_name}
    />,
  );
});

it('renders contact_assignment correctly', () => {
  const item = {
    id: '5',
    _type: 'contact_assignment',
    created_at: date,
  };

  testSnapshotShallow(
    <JourneyItem item={item} myId={myId} personFirstName={person.first_name} />,
  );

  expect(getAssignedToName).toHaveBeenCalledWith(myId, item);
  expect(getAssignedByName).toHaveBeenCalledWith(myId, item);
});

it('renders contact_unassignment correctly', () => {
  const item = {
    id: '6',
    _type: 'contact_unassignment',
    created_at: date,
  };

  testSnapshotShallow(
    <JourneyItem item={item} myId={myId} personFirstName={person.first_name} />,
  );

  expect(getAssignedToName).toHaveBeenCalledWith(myId, item);
});

it('should call ref', () => {
  const item = {
    id: '6',
    _type: 'contact_unassignment',
    created_at: date,
  };

  const instance = renderShallow(
    <JourneyItem item={item} myId={myId} personFirstName={person.first_name} />,
  ).instance();
  const ref = 'test';
  instance.ref(ref);
  expect(instance._view).toEqual(ref);
});
