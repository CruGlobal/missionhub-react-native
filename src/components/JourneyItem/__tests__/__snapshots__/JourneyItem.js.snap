// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders contact_assignment correctly 1`] = `
<Flex
  align="center"
  direction="row"
  style={
    Object {
      "backgroundColor": "#ffffff",
      "paddingRight": 24,
      "paddingVertical": 19,
      "width": 750,
    }
  }
>
  <Flex
    value={1}
  >
    <Icon
      name="statusIcon"
      size={32}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#007398",
            "fontSize": 40,
            "textAlign": "center",
          },
          Object {},
        ]
      }
      type="MissionHub"
    />
  </Flex>
  <Flex
    direction="column"
    style={
      Object {
        "borderLeftColor": "#E6E8EC",
        "borderLeftWidth": 1,
        "paddingBottom": 5,
        "paddingLeft": 20,
      }
    }
    value={3.5}
  >
    <withI18nextTranslation(DateComponent)
      date="2017-12-06T14:24:52Z"
      format="LL"
      style={
        Object {
          "color": "#A0A2A6",
          "fontSize": 10,
        }
      }
    />
    <MyText
      style={
        Object {
          "color": "#007398",
          "fontSize": 16,
        }
      }
    >
      Test Person was assigned to RogerBilly
    </MyText>
  </Flex>
</Flex>
`;

exports[`renders contact_unassignment correctly 1`] = `
<Flex
  align="center"
  direction="row"
  style={
    Object {
      "backgroundColor": "#ffffff",
      "paddingRight": 24,
      "paddingVertical": 19,
      "width": 750,
    }
  }
>
  <Flex
    value={1}
  >
    <Icon
      name="statusIcon"
      size={32}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#007398",
            "fontSize": 40,
            "textAlign": "center",
          },
          Object {},
        ]
      }
      type="MissionHub"
    />
  </Flex>
  <Flex
    direction="column"
    style={
      Object {
        "borderLeftColor": "#E6E8EC",
        "borderLeftWidth": 1,
        "paddingBottom": 5,
        "paddingLeft": 20,
      }
    }
    value={3.5}
  >
    <withI18nextTranslation(DateComponent)
      date="2017-12-06T14:24:52Z"
      format="LL"
      style={
        Object {
          "color": "#A0A2A6",
          "fontSize": 10,
        }
      }
    />
    <MyText
      style={
        Object {
          "color": "#007398",
          "fontSize": 16,
        }
      }
    >
      Test Person was unassigned from Roger
    </MyText>
  </Flex>
</Flex>
`;

exports[`renders interaction correctly 1`] = `
<Flex
  align="center"
  direction="row"
  style={
    Object {
      "backgroundColor": "#ffffff",
      "paddingRight": 24,
      "paddingVertical": 19,
      "width": 750,
    }
  }
>
  <Flex
    value={1}
  >
    <Icon
      name="commentIcon"
      size={32}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#007398",
            "fontSize": 40,
            "textAlign": "center",
          },
          Object {
            "fontSize": 32,
          },
        ]
      }
      type="MissionHub"
    />
  </Flex>
  <Flex
    direction="column"
    style={
      Object {
        "borderLeftColor": "#E6E8EC",
        "borderLeftWidth": 1,
        "paddingBottom": 5,
        "paddingLeft": 20,
      }
    }
    value={3.5}
  >
    <withI18nextTranslation(DateComponent)
      date="2017-12-06T14:24:52Z"
      format="LL"
      style={
        Object {
          "color": "#A0A2A6",
          "fontSize": 10,
        }
      }
    />
    <MyText
      style={
        Object {
          "color": "#007398",
          "fontSize": 16,
        }
      }
    >
      Comment
    </MyText>
    <MyText
      style={
        Object {
          "fontSize": 14,
        }
      }
    >
      Test Interaction
    </MyText>
  </Flex>
</Flex>
`;

exports[`renders survey correctly 1`] = `
<Flex
  align="center"
  direction="row"
  style={
    Object {
      "backgroundColor": "#ffffff",
      "paddingRight": 24,
      "paddingVertical": 19,
      "width": 750,
    }
  }
>
  <Flex
    value={1}
  >
    <Icon
      name="surveyIcon"
      size={32}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#007398",
            "fontSize": 40,
            "textAlign": "center",
          },
          Object {},
        ]
      }
      type="MissionHub"
    />
  </Flex>
  <Flex
    direction="column"
    style={
      Object {
        "borderLeftColor": "#E6E8EC",
        "borderLeftWidth": 1,
        "paddingBottom": 5,
        "paddingLeft": 20,
      }
    }
    value={3.5}
  >
    <withI18nextTranslation(DateComponent)
      date="2017-12-06T14:24:52Z"
      format="LL"
      style={
        Object {
          "color": "#A0A2A6",
          "fontSize": 10,
        }
      }
    />
    <MyText
      style={
        Object {
          "color": "#007398",
          "fontSize": 16,
        }
      }
    >
      Test Survey
    </MyText>
    <Flex
      direction="column"
      key="1"
    >
      <MyText
        style={
          Object {
            "color": "#A0A2A6",
            "fontSize": 14,
            "marginTop": 8,
          }
        }
      >
        Question 1
      </MyText>
      <MyText
        style={
          Object {
            "fontSize": 14,
          }
        }
      >
        Answer 1
      </MyText>
    </Flex>
    <Flex
      direction="column"
      key="2"
    >
      <MyText
        style={
          Object {
            "color": "#A0A2A6",
            "fontSize": 14,
            "marginTop": 8,
          }
        }
      >
        Question 2
      </MyText>
      <MyText
        style={
          Object {
            "fontSize": 14,
          }
        }
      >
        Answer 2
      </MyText>
    </Flex>
    <Flex
      direction="column"
      key="3"
    >
      <MyText
        style={
          Object {
            "color": "#A0A2A6",
            "fontSize": 14,
            "marginTop": 8,
          }
        }
      >
        Question 3
      </MyText>
      <MyText
        style={
          Object {
            "fontSize": 14,
          }
        }
      >
        Answer 3
      </MyText>
    </Flex>
  </Flex>
</Flex>
`;

exports[`stage is rendered correctly with old stage for a contact 1`] = `
<Flex
  align="center"
  direction="row"
  style={
    Object {
      "backgroundColor": "#ffffff",
      "paddingRight": 24,
      "paddingVertical": 19,
      "width": 750,
    }
  }
>
  <Flex
    value={1}
  >
    <Icon
      name="journeyIcon"
      size={32}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#007398",
            "fontSize": 40,
            "textAlign": "center",
          },
          Object {},
        ]
      }
      type="MissionHub"
    />
  </Flex>
  <Flex
    direction="column"
    style={
      Object {
        "borderLeftColor": "#E6E8EC",
        "borderLeftWidth": 1,
        "paddingBottom": 5,
        "paddingLeft": 20,
      }
    }
    value={3.5}
  >
    <withI18nextTranslation(DateComponent)
      date="2017-12-06T14:24:52Z"
      format="LL"
      style={
        Object {
          "color": "#A0A2A6",
          "fontSize": 10,
        }
      }
    />
    <MyText
      style={
        Object {
          "color": "#007398",
          "fontSize": 16,
        }
      }
    >
      Uninterested to Curious
    </MyText>
    <MyText
      style={
        Object {
          "fontSize": 14,
        }
      }
    >
      Test Person changed from Uninterested to Curious
    </MyText>
  </Flex>
</Flex>
`;

exports[`stage is rendered correctly with old stage for self 1`] = `
<Flex
  align="center"
  direction="row"
  style={
    Object {
      "backgroundColor": "#ffffff",
      "paddingRight": 24,
      "paddingVertical": 19,
      "width": 750,
    }
  }
>
  <Flex
    value={1}
  >
    <Icon
      name="journeyIcon"
      size={32}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#007398",
            "fontSize": 40,
            "textAlign": "center",
          },
          Object {},
        ]
      }
      type="MissionHub"
    />
  </Flex>
  <Flex
    direction="column"
    style={
      Object {
        "borderLeftColor": "#E6E8EC",
        "borderLeftWidth": 1,
        "paddingBottom": 5,
        "paddingLeft": 20,
      }
    }
    value={3.5}
  >
    <withI18nextTranslation(DateComponent)
      date="2017-12-06T14:24:52Z"
      format="LL"
      style={
        Object {
          "color": "#A0A2A6",
          "fontSize": 10,
        }
      }
    />
    <MyText
      style={
        Object {
          "color": "#007398",
          "fontSize": 16,
        }
      }
    >
      Uninterested to Curious
    </MyText>
    <MyText
      style={
        Object {
          "fontSize": 14,
        }
      }
    >
      You changed from Uninterested to Curious
    </MyText>
  </Flex>
</Flex>
`;

exports[`stage is rendered correctly without old stage for a contact 1`] = `
<Flex
  align="center"
  direction="row"
  style={
    Object {
      "backgroundColor": "#ffffff",
      "paddingRight": 24,
      "paddingVertical": 19,
      "width": 750,
    }
  }
>
  <Flex
    value={1}
  >
    <Icon
      name="journeyIcon"
      size={32}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#007398",
            "fontSize": 40,
            "textAlign": "center",
          },
          Object {},
        ]
      }
      type="MissionHub"
    />
  </Flex>
  <Flex
    direction="column"
    style={
      Object {
        "borderLeftColor": "#E6E8EC",
        "borderLeftWidth": 1,
        "paddingBottom": 5,
        "paddingLeft": 20,
      }
    }
    value={3.5}
  >
    <withI18nextTranslation(DateComponent)
      date="2017-12-06T14:24:52Z"
      format="LL"
      style={
        Object {
          "color": "#A0A2A6",
          "fontSize": 10,
        }
      }
    />
    <MyText
      style={
        Object {
          "color": "#007398",
          "fontSize": 16,
        }
      }
    >
      Curious
    </MyText>
    <MyText
      style={
        Object {
          "fontSize": 14,
        }
      }
    >
      You added Test Person to MissionHub as Curious
    </MyText>
  </Flex>
</Flex>
`;

exports[`stage is rendered correctly without old stage for self 1`] = `
<Flex
  align="center"
  direction="row"
  style={
    Object {
      "backgroundColor": "#ffffff",
      "paddingRight": 24,
      "paddingVertical": 19,
      "width": 750,
    }
  }
>
  <Flex
    value={1}
  >
    <Icon
      name="journeyIcon"
      size={32}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#007398",
            "fontSize": 40,
            "textAlign": "center",
          },
          Object {},
        ]
      }
      type="MissionHub"
    />
  </Flex>
  <Flex
    direction="column"
    style={
      Object {
        "borderLeftColor": "#E6E8EC",
        "borderLeftWidth": 1,
        "paddingBottom": 5,
        "paddingLeft": 20,
      }
    }
    value={3.5}
  >
    <withI18nextTranslation(DateComponent)
      date="2017-12-06T14:24:52Z"
      format="LL"
      style={
        Object {
          "color": "#A0A2A6",
          "fontSize": 10,
        }
      }
    />
    <MyText
      style={
        Object {
          "color": "#007398",
          "fontSize": 16,
        }
      }
    >
      Curious
    </MyText>
    <MyText
      style={
        Object {
          "fontSize": 14,
        }
      }
    >
      You added yourself to MissionHub as Curious
    </MyText>
  </Flex>
</Flex>
`;

exports[`step is rendered correctly with comment 1`] = `
<Flex
  align="center"
  direction="row"
  style={
    Object {
      "backgroundColor": "#ffffff",
      "paddingRight": 24,
      "paddingVertical": 19,
      "width": 750,
    }
  }
>
  <Flex
    value={1}
  >
    <Icon
      name="stepsIcon"
      size={32}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#007398",
            "fontSize": 40,
            "textAlign": "center",
          },
          Object {},
        ]
      }
      type="MissionHub"
    />
  </Flex>
  <Flex
    direction="column"
    style={
      Object {
        "borderLeftColor": "#E6E8EC",
        "borderLeftWidth": 1,
        "paddingBottom": 5,
        "paddingLeft": 20,
      }
    }
    value={3.5}
  >
    <withI18nextTranslation(DateComponent)
      date="2017-12-06T14:24:52Z"
      format="LL"
      style={
        Object {
          "color": "#A0A2A6",
          "fontSize": 10,
        }
      }
    />
    <MyText
      style={
        Object {
          "color": "#007398",
          "fontSize": 16,
        }
      }
    >
      Completed Step of Faith
    </MyText>
    <MyText
      style={
        Object {
          "fontSize": 14,
        }
      }
    >
      Test Step

test comment on completed step
    </MyText>
  </Flex>
</Flex>
`;

exports[`step is rendered correctly with pathway stage 1`] = `
<Flex
  align="center"
  direction="row"
  style={
    Object {
      "backgroundColor": "#ffffff",
      "paddingRight": 24,
      "paddingVertical": 19,
      "width": 750,
    }
  }
>
  <Flex
    value={1}
  >
    <Icon
      name="stepsIcon"
      size={32}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#007398",
            "fontSize": 40,
            "textAlign": "center",
          },
          Object {},
        ]
      }
      type="MissionHub"
    />
  </Flex>
  <Flex
    direction="column"
    style={
      Object {
        "borderLeftColor": "#E6E8EC",
        "borderLeftWidth": 1,
        "paddingBottom": 5,
        "paddingLeft": 20,
      }
    }
    value={3.5}
  >
    <withI18nextTranslation(DateComponent)
      date="2017-12-06T14:24:52Z"
      format="LL"
      style={
        Object {
          "color": "#A0A2A6",
          "fontSize": 10,
        }
      }
    />
    <MyText
      style={
        Object {
          "color": "#007398",
          "fontSize": 16,
        }
      }
    >
      Completed Guiding Step of Faith
    </MyText>
    <MyText
      style={
        Object {
          "fontSize": 14,
        }
      }
    >
      Test Step
    </MyText>
  </Flex>
</Flex>
`;

exports[`step is rendered correctly without comment 1`] = `
<Flex
  align="center"
  direction="row"
  style={
    Object {
      "backgroundColor": "#ffffff",
      "paddingRight": 24,
      "paddingVertical": 19,
      "width": 750,
    }
  }
>
  <Flex
    value={1}
  >
    <Icon
      name="stepsIcon"
      size={32}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "color": "#007398",
            "fontSize": 40,
            "textAlign": "center",
          },
          Object {},
        ]
      }
      type="MissionHub"
    />
  </Flex>
  <Flex
    direction="column"
    style={
      Object {
        "borderLeftColor": "#E6E8EC",
        "borderLeftWidth": 1,
        "paddingBottom": 5,
        "paddingLeft": 20,
      }
    }
    value={3.5}
  >
    <withI18nextTranslation(DateComponent)
      date="2017-12-06T14:24:52Z"
      format="LL"
      style={
        Object {
          "color": "#A0A2A6",
          "fontSize": 10,
        }
      }
    />
    <MyText
      style={
        Object {
          "color": "#007398",
          "fontSize": 16,
        }
      }
    >
      Completed Step of Faith
    </MyText>
    <MyText
      style={
        Object {
          "fontSize": 14,
        }
      }
    >
      Test Step
    </MyText>
  </Flex>
</Flex>
`;
